File: dataset/Vanilla/GEMINI_gemini-2.5-pro/Task36.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.io.File;

public class Task36 {

    /**
     * Reads a file specified by the file path and prints its contents to the console.
     *
     * @param filePath The path to the file to be read.
     */
    public static void readFileAndPrint(String filePath) {
        System.out.println("Reading file: " + filePath);
        File file = new File(filePath);
        if (!file.exists()) {
            System.err.println("Error: File not found at '" + filePath + "'");
            return;
        }

        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
            String line;
            boolean empty = true;
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
                empty = false;
            }
            if(empty){
                 System.out.println("(File is empty)");
            }
        } catch (IOException e) {
            System.err.println("Error reading file: " + e.getMessage());
        }
    }

    /**
     * Helper function to create a test file with specified content.
     */
    private static void createTestFile(String fileName, String content) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
            writer.write(content);
        } catch (IOException e) {
            System.err.println("Failed to create test file: " + fileName);
            e.printStackTrace();
        }
    }

    public static void main(String[] args) {
        // --- Test Case 1: Read a simple, single-line file ---
        System.out.println("--- Test Case 1: Simple File ---");
        String file1 = "java_test1.txt";
        createTestFile(file1, "Hello, World from Java!");
        readFileAndPrint(file1);
        System.out.println("-------------------------------------\n");

        // --- Test Case 2: Read a multi-line file ---
        System.out.println("--- Test Case 2: Multi-line File ---");
        String file2 = "java_test2.txt";
        createTestFile(file2, "This is line one.\nThis is line two.\nAnd this is the third line.");
        readFileAndPrint(file2);
        System.out.println("-------------------------------------\n");

        // --- Test Case 3: Read an empty file ---
        System.out.println("--- Test Case 3: Empty File ---");
        String file3 = "java_test3_empty.txt";
        createTestFile(file3, "");
        readFileAndPrint(file3);
        System.out.println("-------------------------------------\n");

        // --- Test Case 4: Attempt to read a non-existent file ---
        System.out.println("--- Test Case 4: Non-existent File ---");
        readFileAndPrint("nonexistentfile.txt");
        System.out.println("-------------------------------------\n");

        // --- Test Case 5: Read a file with special characters ---
        System.out.println("--- Test Case 5: Special Characters File ---");
        String file4 = "java_test4_special.txt";
        createTestFile(file4, "Special chars: ñ, é, ö, ü, ç, α, β, γ. 😊🎉");
        readFileAndPrint(file4);
        System.out.println("-------------------------------------\n");

        // Cleanup test files
        new File(file1).delete();
        new File(file2).delete();
        new File(file3).delete();
        new File(file4).delete();
    }
}

File: dataset/Vanilla/GEMINI_gemini-2.5-pro/Task36.py
import os

def read_file_and_print(file_path):
    """
    Reads a file specified by the file path and prints its contents to the console.
    
    :param file_path: The path to the file to be read.
    """
    print(f"Reading file: {file_path}")
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
            if not content:
                print("(File is empty)")
            else:
                # Use print() which adds a newline, so no need for end=''
                # if the file content itself doesn't end with a newline
                print(content, end='')
    except FileNotFoundError:
        print(f"Error: File not found at '{file_path}'")
    except Exception as e:
        print(f"An error occurred: {e}")

def create_test_file(filename, content):
    """Helper function to create a test file."""
    try:
        with open(filename, 'w', encoding='utf-8') as f:
            f.write(content)
    except Exception as e:
        print(f"Failed to create test file {filename}: {e}")

# Main execution block with test cases
if __name__ == "__main__":
    # --- Test Case 1: Read a simple, single-line file ---
    print("--- Test Case 1: Simple File ---")
    file1 = "py_test1.txt"
    create_test_file(file1, "Hello, World from Python!")
    read_file_and_print(file1)
    print("\n-------------------------------------\n")

    # --- Test Case 2: Read a multi-line file ---
    print("--- Test Case 2: Multi-line File ---")
    file2 = "py_test2.txt"
    create_test_file(file2, "This is line one.\nThis is line two.\nAnd this is the third line.")
    read_file_and_print(file2)
    print("\n-------------------------------------\n")

    # --- Test Case 3: Read an empty file ---
    print("--- Test Case 3: Empty File ---")
    file3 = "py_test3_empty.txt"
    create_test_file(file3, "")
    read_file_and_print(file3)
    print("\n-------------------------------------\n")

    # --- Test Case 4: Attempt to read a non-existent file ---
    print("--- Test Case 4: Non-existent File ---")
    read_file_and_print("nonexistentfile.txt")
    print("-------------------------------------\n")

    # --- Test Case 5: Read a file with special characters ---
    print("--- Test Case 5: Special Characters File ---")
    file4 = "py_test4_special.txt"
    create_test_file(file4, "Special chars: ñ, é, ö, ü, ç, α, β, γ. 😊🎉")
    read_file_and_print(file4)
    print("\n-------------------------------------\n")
    
    # Cleanup test files
    os.remove(file1)
    os.remove(file2)
    os.remove(file3)
    os.remove(file4)